File: 2_practice/brute_force_2.cpp
#include <iostream>
using namespace std;

int main()
{
  int A[10000], N, S, sum = 0;

  cin >> N >> S;
  for(int i = 1; i <= N; i++) cin >> A[i];

  for(int i = 1; i <= N; i++) sum = sum + A[i];
  
  if(sum % S == 0) cout << "yes" << endl;
  else cout << "no" << endl;
}

File: ruby/2_practice/brute_force_2.rb
require "pry"
## 小さい具体例で何ができたらいいか考えてみる
# ary[0] + ary[1] == s
# ary[0] + ary[1] + ary[2] == s
# ary[0] + ary[2] == s
# ary[1] + ary[2] == s
def main
  n, s = gets.split(" ").map(&:to_i)
  ary = gets.split(" ").map(&:to_i)
  answer = false

  (0..n -1).each do |i|
    sum = ary[i]
    (i + 1..n - 1).each do |j|
      sum = sum + ary[j]
      count = true if sum == s
    end
  end

  puts answer == true ? 'Yes' : 'No'
end

main
